buildSrc/src/main/groovy/com/carrotsearch/gradle/junit4/RandomizedTestingPlugin.groovy

```diff
@@ -45,29 +65,33 @@ class RandomizedTestingPlugin implements Plugin<Project> {
     }
 
     static void replaceTestTask(TaskContainer tasks) {
-        Test oldTestTask = tasks.findByPath('test')
-        if (oldTestTask == null) {
+        // Gradle 4.8 introduced lazy tasks, thus we deal both with the `test` task as well as it's provider
+        // https://github.com/gradle/gradle/issues/5730#issuecomment-398822153
+        // since we can't be sure if the task was ever realized, we remove both the provider and the task
+        TaskProvider<Test> oldTestProvider
+        try {
+            oldTestProvider = tasks.getByNameLater(Test, 'test')
+        }
+        catch (UnknownTaskException why) {
```

Comment:
Could you rename the exception to something more descriptive? `why` makes me think we don't know why the exception is thrown. Also it'd be lovely if you'd move the catch up to the `}` line.